wolfcrypt/src/port/maxim/maxq10xx.c

```diff
@@ -0,0 +1,3376 @@
+ */
+
+#ifdef HAVE_CONFIG_H
+    #include <config.h>
+#endif
+
+#include <wolfssl/wolfcrypt/settings.h>
+
+#if defined(WOLFSSL_MAXQ1065) ||  defined(WOLFSSL_MAXQ108X)
+
+#include <wolfssl/wolfcrypt/port/maxim/maxq10xx.h>
+
+#ifdef NO_INLINE
+    #include <wolfssl/wolfcrypt/misc.h>
+#else
+    #define WOLFSSL_MISC_INCLUDED
+    #include <wolfcrypt/src/misc.c>
+#endif
+
+#include <stdint.h>
+#include <wolfssl/internal.h>
+#include <wolfssl/wolfcrypt/cryptocb.h>
+#include <wolfssl/wolfcrypt/error-crypt.h>
+#include <wolfssl/wolfcrypt/logging.h>
+#include <wolfssl/wolfcrypt/port/maxim/MXQ_API.h>
+
+#ifdef MAXQ_DEBUG
+void dbg_dumphex(const char *identifier, const uint8_t* pdata, uint32_t plen);
+#else
+#define dbg_dumphex(identifier, pdata, plen)
+#endif /* MAXQ_DEBUG */
+
+#if defined(USE_WINDOWS_API)
+# define maxq_CryptHwMutexTryLock()	(0)
+#endif
+
+#define AES_KEY_ID_START      (0x2000)
+#define AES_KEY_ID_MAX_NUM    (32)
+#define ECC_KEY_ID_START      (AES_KEY_ID_START + AES_KEY_ID_MAX_NUM)
+#define ECC_KEY_ID_MAX_NUM    (32)
+
+#define TEMP_KEY_ID_START     (0)
+#if defined(WOLFSSL_MAXQ108X)
+#define TEMP_KEY_ID_MAX_NUM   (16)
+#else
+#define TEMP_KEY_ID_MAX_NUM   (2)
+#endif
+
+#define PUBKEY_IMPORT_OBJID    0x1000
+#define ROOT_CA_CERT_OBJ_ID    0x1003
+#define DEVICE_CERT_OBJ_ID     0x1002
+#define DEVICE_KEY_PAIR_OBJ_ID 0x1004
+#define PSK_OBJ_ID             0x1234
+#define K_CHUNKSIZE            2032
+#define K_CIPHER_BLOCKSIZE     16
+#define ECC256_KEYSIZE         32
+
+#ifdef WOLFSSL_MAXQ108X
+
+#define PSK_KID (0x1235)
+static unsigned char tls13active          =  0;
+static unsigned char tls13early           =  0;
+
+static int tls13_dh_obj_id                = -1;
+static int tls13_ecc_obj_id               = -1;
+static int tls13_hs_early_secret_obj_id   = -1;
+static int tls13_binder_key_obj_id        = -1;
+static int tls13_shared_secret_obj_id     = -1;
+static int tls13_early_secret_obj_id      = -1;
+static int tls13_derived_secret_obj_id    = -1;
+static int tls13_handshake_secret_obj_id  = -1;
+static int tls13_master_secret_obj_id     = -1;
+static int tls13_client_secret_obj_id     = -1;
+static int tls13_server_secret_obj_id     = -1;
+static int tls13_client_hs_key_obj_id     = -1;
+static int tls13_server_hs_key_obj_id     = -1;
+static int tls13_client_app_key_obj_id    = -1;
+static int tls13_server_app_key_obj_id    = -1;
+static int tls13_server_finish_obj_id     = -1;
+static int tls13_client_finish_obj_id     = -1;
+static int *tls13_server_key_id           = NULL;
+static int *tls13_client_key_id           = NULL;
+static int tls13_res_master_obj_id        = -1;
+static int tls13_server_cert_id           = -1;
+static int tls13_server_key_algo          = -1;
+static int tls13_server_key_len           = -1;
+
+#endif /* WOLFSSL_MAXQ108X */
+
+/* TEST KEY. This must be changed for production environments!! */
+static const mxq_u1 KeyPairImport[] = {
```

Comment:
Might consider allowing a way for this value to come from an external file based on macro. Something like:

```
#ifdef CUSTOM_MAXQ_KEY
#include "maxq_key.h"
#endif
```